coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,375 @@
+        """
+        fix_suggestions = """
+        Following specific rules about the usage of blank lines: using them
+        only when necessary.
+        """
+
+
+@BlankLine.subaspect
+class BlankLineAfterDeclaration:
+    """Blank lines after declaration are those found after declarations.
+    """
+    class docs:
+        example = """
+        int main ()
+        {
+          int a;
+          return 0;
+        }
+        """
+        example_language = 'C'
+        importance_reason = """
+        Having a specific and reasonable number of blank lines after every
+        block of declarations improves on the readability of the code.
+        """
+        fix_suggestions = """
+        `BlankLintAfterDeclaration` issues can be fixed specifying (and of
+        course using) a reasonable number of blank lines to use after block
+        declaration.
+        """
+    blank_lines_after_declarations = Taste[int](
+        'Represents the number of blank lines after declarations',
+        (0, 1, 2), default=0)
+
+
+@BlankLine.subaspect
+class BlankLineAfterProcedure:
+    """Blank lines after procedures are those found after procedures.
+    """
+    class docs:
+        example = """
+        void proc(void){
+            printf("this does nothing");
+        }
+
+        int add(float a, float b){
+            return a + b;
+        }
+        """
+        example_language = 'C++'
+        importance_reason = """
+        Having a specific and reasonable number of blank lines after every
+        procedures improves on the readability of the code.
+        """
+        fix_suggestions = """
+        `BlankLintAfterProcedure` issues can be fixed specifying (and of
+        course using) a reasonable number of blank lines to use after
+        procedures' definition.
+        """
+    blank_lines_after_procedures = Taste[int](
+        'Represents the number of blank lines to use after a procedure or a'
+        ' function', (0, 1, 2), default=1)
+
+
+@BlankLine.subaspect
+class BlankLineAfterClass:
+    """Blank lines after classes are those found after classes' definitions.
+    """
+    class docs:
+        example = """
+        class SomeClass:
+            def __init__(self):
+                raise RuntimeError('Never instantiate this class')
+
+
+        def func():
+            pass
+        """
+        example_language = 'python'
+        importance_reason = """
+        Having a specific number of blank lines after every classes'
+        definitions declarations improves on the readability of the code.
+        """
+        fix_suggestions = """
+        """
+    blank_lines_after_class = Taste[int](
+        'Represents the number of blank lines to use after a class'
+        'definition.', (1, 2), default=2)
+
+
+@BlankLine.subaspect
+class NewlineAtEOF:
+    """Illustrate the end of file character.
+
+    A newline character (usually '\\\\n', aka CR or CRLF) at found at the end
```

Comment:
"at found at" needs improvement.  "is found at" is slightly better , but not good enough imo.